hugo/content/blog/2021-12-07-new-versioning-scheme/index.md

```diff
@@ -0,0 +1,54 @@
+---
+title: New versioning scheme
+author: Robert Adam
+date: 2021-12-07
+categories:
+  - News
+---
+Mumble traditionally used a versioning scheme of the form major.minor.patch where the patch version number was incremented for every bug-fix release
+of a stable release series (e.g. 1.3.0 -> 1.3.1). Most notably though, the patch version was always starting at zero for a new stable release series
+(e.g. 1.2.0, 1.3.0, etc.). This will no longer be the case.
+
+<!--more-->
+
+## Background
+
+Mumble was always using the [semantic versioning scheme](https://semver.org/) which generally makes it really easy to know what to expect when
+switching from one version to another. While this is working great, using only the three common version numbers (major.minor.patch) poses a problem
+when considering snapshot builds. A snapshot build is essentially a beta-release for an upcoming release. Thus it has to use the same version numbers
+as that upcoming release. Thus, in order to get a unique versioning scheme, we had to make use of the fourth version number, that semantic versioning
+allows for. We called this the _build number_. This build number was just a number that was incremented with every build of the Mumble binaries that
+we performed. Thus a version could now look like this: 1.4.0.62 and the stable release would then simply use whatever build number was next in the
+queue.
+
+## The problem
+
+In theory that worked great and solved the issue at hand: we now had a unique version for every release (be that a stable or a snapshot release).
+However, as it turns out Windows hasn't arrived in the 21st century yet: it restricts programs to use a three-component versioning scheme. While it is

```

Comment:
Is this a Windows thing? A MSI installer thing?

Reply:
I don't like the aggressive wording. I though it could be worded better. Even with voicing frustration or shortcomings if we want to do so explicitly.

Reply:
I think we don't even know where exactly the issue is rooted.

Reply:
Indeed, for all we know it could be an issue in WiX rather than the MSI subsystem.